compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/SpecialBridgeMethods.kt

```diff
@@ -103,11 +104,16 @@ class SpecialBridgeMethods(val context: CommonBackendContext) {
         makeDescription(KotlinBuiltIns.FQ_NAMES.mutableList, "removeAt", 1) to BuiltInWithDifferentJvmName(needsGenericSignature = true)
     )
 
-    fun findSpecialWithOverride(irFunction: IrSimpleFunction): Pair<IrSimpleFunction, SpecialMethodWithDefaultInfo>? {
-        irFunction.allOverridden().forEach { overridden ->
-            val description = overridden.toDescription()
-            SPECIAL_METHODS_WITH_DEFAULTS_MAP[description]?.let {
-                return Pair(overridden, it)
+    fun findSpecialWithOverride(
+        irFunction: IrSimpleFunction,
+        includeSelf: Boolean = false
+    ): Pair<IrSimpleFunction, SpecialMethodWithDefaultInfo>? {
+        irFunction.allOverridden(includeSelf).forEach { overridden ->
+            if (overridden.parent is IrClass) {
```

Comment:
How is it possible to have an override with a parent not inside an `IrClass`?

Reply:
It isn't, but if `includeSelf` is `true` we're also looking at the `irFunction` which might be an external top-level declaration or intrinsic.

I've hoisted the check outside of the `forEach` to make a code a bit cleaner.